### PR TITLE
Make the AI play more aggressively when no threats

### PR DESCRIPTION
**Pull request by demonoidv**
I guess the title is self-explanatory, however, the idea behind this PR is to make the AI play a bit more aggressively when there are no imminent threats at depth 0 (meaning the computation of the potential threats is made only on the initial node in order not to narrow down the perf too much).
To do so, we apply more weight to the defense fscore if Player is threatened at depth 0 which should make the AI choose a more defensive move than if there are no threats at all. Obviously, the wanted side effect of this is that as long as there are no imminent threats, the AI should attack as much as possible.

I'm totally open to discuss this approach and suggestions are more than welcome.

### CODE REVIEW

**Review by arlaine4 — APPROVED** [review]

File: src/algorithm/mod.rs
Comment (Ragarnoy) [inline src/algorithm/mod.rs:34]: normal?

File: src/algorithm/mod.rs
Comment (Ragarnoy) [inline src/algorithm/mod.rs:262]: normal?

**Review by Ragarnoy — APPROVED** [review]
Ok except for 2 fixmes?

File: src/algorithm/mod.rs
Comment (demonoidv) [reply src/algorithm/mod.rs:34]: Yup, not an issue, just a reminder for good practice. I still need it to be public at this point but we should turn it into a private method when we'll be ready.

**Comment by demonoidv** [comment]
> Ok except for 2 fixmes?

I think you shouldn't approve the changes if you expect an answer before I merge the PR because if you do, I assume that you are OK with those changes and will merge the PR as soon as possible, is that what you want here? (obviously, no problem if this is what you meant to do, it's perfectly acceptable, I just want to be sure that you have nothing else to say about this PR)

**Comment by Ragarnoy** [comment]
> 
> 
> > Ok except for 2 fixmes?
> 
> I think you shouldn't approve the changes if you expect an answer before I merge the PR because if you do, I assume that you are OK with those changes and will merge the PR as soon as possible, is that what you want here? (obviously, no problem if this is what you meant to do, it's perfectly acceptable, I just want to be sure that you have nothing else to say about this PR)

Yep

**Pull request merged**
